app/Http/Controllers/SettingController.php

```diff
@@ -2,9 +2,86 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\StoreSettingRequest;
+use App\Models\Setting;
+use App\Models\User;
 use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
 
 class SettingController extends Controller
 {
-    //
+    public function index()
+    {
+        // get Sam to consolidate this, it could be improved, maybe a different way instead of get()?
+        return view('settings.index',[
+            'settings' => auth()->user()->settings()->get()
+        ]);
+
+    }
+
+    public static function create(User $user, string $key, string $value)
+    {
+        //creates a user setting from the parameters. use in code only. will be redundant probably.
+
+        $user->settings()->create([
+            'user_id' => $user->id,
+            'key' => $key,
+            'value' => $value
+        ]);
+
+    }
+
+    public function store(StoreSettingRequest $request)
+    {
+        //creates a user setting for the currently logged in user /w validation
+
+
+        //TODO: complete authorization in StoreSettingRequest
+        $validated = $request->validated();
+
+        auth()->user()->settings()->create(
+            $validated
+        );
```

Comment:
To improve readability, I would separate the user into their own variable and then keep it on the same line. For example:

```php
$user = auth()->user();

$user->settings()->create($validated);
```